restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -740,7 +753,11 @@ void deleteDeadLeaves(
     }
 
     // Fire this off in a future
-    dataStore.batch(queries, ConsistencyLevel.LOCAL_QUORUM);
+    if (useLoggedBatches) {
+      dataStore.batch(queries, ConsistencyLevel.LOCAL_QUORUM).join();
+    } else {
+      dataStore.unloggedBatch(queries, ConsistencyLevel.LOCAL_QUORUM).join();
+    }
```

Comment:
nit: Extract this logic into a method? It is repeated 3 times in this class :shrug:

Reply:
I want to, but I really can't have CI run again 😬

Reply:
If it fails for any reason I will make this change